# Copyright (c) 2013 LG Electronics, Inc.

# Busybox implementation only supports head -1 option if ENABLE_INCLUDE_SUSv2
# or ENABLE_FEATURE_FANCY_HEAD configuration options are enabled. Also the -and
# option for find is only supported if ENABLE_DESKTOP configuration option is
# enabled. These configuration options are not enabled in build-webos, which is
# why this patch is needed.

diff -ru orig-lsb-release-1.4/lsb_release lsb-release-1.4/lsb_release
--- orig-lsb-release-1.4/lsb_release	2013-03-13 09:13:56.518509117 +0200
+++ lsb-release-1.4/lsb_release	2013-03-13 11:50:03.765119076 +0200
@@ -209,7 +209,7 @@
             || [ -n "$(echo $DISTRIB_DESCRIPTION |                            \
                        sed -e "s/.*$DESCSTR_DELI.*//")" ]
         then
-            TMP_DISTRIB_DESC=$(head -1 $FILENAME 2>/dev/null)
+            TMP_DISTRIB_DESC=$(head -n 1 $FILENAME 2>/dev/null)
             [ -z "$DISTRIB_DESCRIPTION" ]                                     \
                 && DISTRIB_DESCRIPTION=$TMP_DISTRIB_DESC
         else
@@ -249,10 +249,10 @@
         then
             CHECKFIRST=$(find $INFO_ROOT/ -maxdepth 1                         \
                    -name \*$INFO_DISTRIB_SUFFIX                               \
-                   -and ! -name $INFO_LSB_FILE                                \
-                   -and -type f                                               \
+                   -type f                                                    \
                    2>/dev/null                                                \
-                   | head -1 ) # keep one of the files found (if many)
+                   | grep -v $INFO_LSB_FILE                                   \
+                   | head -n 1 ) # keep one of the files found (if many)
         fi
         InitDistribInfo $CHECKFIRST
     fi
